pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,598 @@
+    return(worst_results)
+
+
+def sdm_pretty_print_gold_dataset(json_filepath=None, csv_filepath=None):
+    """
+    TODO
+    Returns the dataset as a human-readale, comma-separated-value (csv) string.
+    Optionally writes result in lossy csv file.
+    """
+
+    # Load gold dataset (converted)
+    gold_dataset = sdm_load_gold_dataset_converted(json_filepath=json_filepath)
+
+    num_pts = len(gold_dataset["devices"][0]["iv_curves"][0]["v_gold"])
+
+    # Initialize output
+    gold_dataset_csv = gold_dataset["timestamp_utc_iso"] + \
+        ", " + gold_dataset["dataset_version"] + "\n\n"
+    gold_dataset_csv = gold_dataset_csv + "G (W/m^2), T (degC)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        "r_sh (Ohm), r_s (Ohm), nNsVth (V), i_0 (A), i_l (A)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["v_gold_" + str(idx) +
+                   " (V)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["i_gold_" + str(idx) +
+                   " (A)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + "\n\n"
+
+    for device in gold_dataset["devices"]:
+        for iv_curve in device["iv_curves"]:
+            # Load G, T, and 5 single diode model parameters for each I-V curve
+            # Floats are converted from lossless, interchangeable hex format
+            gold_dataset_csv = gold_dataset_csv + ", ".join(map(str,
+                                                                [
+                                                                    iv_curve["G"],
+                                                                    iv_curve["T"]
+                                                                ])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(map(str,
+                                                                [
+                                                                    iv_curve["r_sh"],
+                                                                    iv_curve["r_s"],
+                                                                    iv_curve["nNsVth"],
+                                                                    iv_curve["i_0"],
+                                                                    iv_curve["i_l"]
+                                                                ])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["v_gold"].tolist())) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["i_gold"].tolist())) + "\n"
+
+    # When requested, write dataset in csv file, skips empty path strings too
+    if csv_filepath:
+        with open(csv_filepath, 'w') as fp:
+            fp.write(gold_dataset_csv)
+
+    return gold_dataset_csv
+
+
+# TODO This module will be pulled into pvsystem.py and this removed
```

Comment:
I'm not sure I agree with this. I like the way you're doing it now, especially with a readme that lives in this directory.